controllers/users.js

```diff
@@ -113,6 +114,41 @@ const getUsers = async (req, res) => {
       });
     }
 
+    const profile = req.query.profile === "true";
+
+    if (profile) {
```

Comment:
Can we extract this profile fetching logic in a separate method. This method is already very long